plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/utils/PlayAssetDeliveryUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ 	Copyright 2020 Google LLC
+
+ 	Licensed under the Apache License, Version 2.0 (the "License");
+ 	you may not use this file except in compliance with the License.
+ 	You may obtain a copy of the License at
+
+ 		https://www.apache.org/licenses/LICENSE-2.0
+
+ 	Unless required by applicable law or agreed to in writing, software
+ 	distributed under the License is distributed on an "AS IS" BASIS,
+ 	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ 	See the License for the specific language governing permissions and
+ 	limitations under the License.
+*/
+
+package com.google.play.core.godot.assetpacks.utils;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import org.godotengine.godot.Dictionary;
+
+import com.google.android.play.core.assetpacks.AssetPackState;
+
+/**
+ * This class contains all the helper methods for serializing/deserializing custom objects used in
+ * the Play Asset Delivery API. The Java objects are serialized into
+ * org.godotengine.godot.Dictionary, which the Godot runtime can receive. For all helper functions,
+ * null is return upon exception.
+ */
+public class PlayAssetDeliveryUtils {
+
+  private static final String TAG = "PlayAssetDeliveryUtils";
+
+  public static Dictionary constructAssetPackStateDictionary(
+      long bytesDownloaded,
+      int errorCode,
+      String name,
+      int status,
+      long totalBytesToDownload,
+      int transferProgressPercentage) {
+    Dictionary returnDict = new Dictionary();
+    returnDict.put("bytesDownloaded", bytesDownloaded);
+    returnDict.put("errorCode", errorCode);
+    returnDict.put("name", name);
+    returnDict.put("status", status);
+    returnDict.put("totalBytesToDownload", totalBytesToDownload);
+    returnDict.put("transferProgressPercentage", transferProgressPercentage);
+    return returnDict;
+  }
+
+  public static Dictionary convertAssetPackStateToDictionary(AssetPackState assetPackState) {
+    return constructAssetPackStateDictionary(
+        assetPackState.bytesDownloaded(),
+        assetPackState.errorCode(),
+        assetPackState.name(),
+        assetPackState.status(),
+        assetPackState.totalBytesToDownload(),
+        assetPackState.transferProgressPercentage());
+  }
+
+  public static AssetPackState convertDictionaryToAssetPackState(Dictionary dict) {
+    try {
+      AssetPackState packState = new AssetPackStateFromDictionary(dict);
+      return packState;
+    } catch (Exception e) {
```

Comment:
If you do want to return null rather than let the exception bubble up to the caller, I would add a catch block for each exception type you expect, or change AssetPackStateFromDictionary to throw only one type of exception.

Reply:
Now AssetPackStateFromDictionary only throws IllegalArgumentException, and exception here are bubbled up.